lib/log.js

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2020 LG Electronics Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const util = require('util'),
+    async = require('async'),
+    npmlog = require('npmlog'),
+    fs = require('fs'),
+    path = require('path'),
+    novacom = require('./base/novacom'),
+    errHndl = require('./base/error-handler');
+
+(function() {
+
+    const log = npmlog;
+    log.heading = 'log';
+    log.level = 'warn';
+
+    const logLib = {
+
+        /**
+         * @property {Object} log an npm log instance
+         */
+        log: log,
+
+        show: function(options, next) {
+            log.info("log#show");
+            if (typeof next !== 'function') {
+                throw errHndl.changeErrMsg("MISSING_CALLBACK", "next", util.inspect(next));
+            }
+
+            options = options || {};
+            async.series([
+                _generateCmd,
+                _makeSession,
+                function(next){
+                    options.session.run(options.cmd, process.stdin, _onData, process.stderr, next);
+                }
+            ],  function(err) {
+                next(err, null);
+            });
+
+            function _generateCmd(next) {
+                generateCmd(options, next);
+            }
+
+            function _makeSession(next) {
+                makeSession(options, next);
+            }
+
+            function _onData(data) {
+                var str = (Buffer.isBuffer(data))? data.toString():data;
+                console.log(str);
+            }
+        },
+
+        save: function(options, next) {
+            log.info("log#save");
+            if (typeof next !== 'function') {
+                throw errHndl.changeErrMsg("MISSING_CALLBACK", "next", util.inspect(next));
+            }
+
+            options = options || {};
+            async.series([
+                _generateCmd,
+                _makeSession,
+                _createLogFile,
+                function(next){
+                    options.session.run(options.cmd, process.stdin, _onData, process.stderr, next);
+                },
+            ],  function(err) {
+                next(err, null);
+            });
+
+            function _createLogFile(next) {
```

Comment:
It would be better to print the created log file name to user like ares-device capture.

Reply:
Sorry, I don't know what it means...
Please explain verbally.